pyvelox/pyvelox.h

```diff
@@ -504,6 +523,67 @@ static void addVectorBindings(
             std::move(baseVector),
             PyVeloxContext::getSingletonInstance().pool());
       });
+
+  m.def(
+      "row_vector",
+      [](std::vector<std::string>& children,
+         std::vector<VectorPtr>& values,
+         const std::optional<py::dict>& nullabilityDict) {
+        std::vector<std::shared_ptr<const Type>> childTypes;
+        childTypes.reserve(values.size());
+        size_t vectorSize = values.size() > 0 ? values[0]->size() : 0;
+        for (int i = 0; i < values.size(); i++) {
+          // choose child with smallest size to calculate the vector size
+          if (i > 0 && values[i]->size() < vectorSize) {
+            vectorSize = values[i]->size();
+          }
+          childTypes.push_back(values[i]->type());
+        }
+        auto rowType = ROW(std::move(children), std::move(childTypes));
+
+        BufferPtr nullabilityBuffer = nullptr;
+        if (nullabilityDict.has_value()) {
+          auto nullabilityValues = nullabilityDict.value();
+          nullabilityBuffer = AlignedBuffer::allocate<bool>(
+              values.size(),
+              PyVeloxContext::getSingletonInstance().pool(),
+              true);
+          for (const auto&& item : nullabilityValues) {
+            auto row = item.first;
+            auto nullability = item.second;
+            if (!py::isinstance<py::int_>(row) ||
+                !py::isinstance<py::bool_>(nullability)) {
+              throw py::type_error(
+                  "nullability must be a dictionary, rowId in int and nullability in boolean");
+            }
+            int rowId = py::cast<int>(row);
+            bool nullabilityVal = py::cast<bool>(nullability);
+            bits::setBit(
+                nullabilityBuffer->asMutable<uint64_t>(),
+                rowId,
+                bits::kNull ? nullabilityVal : !nullabilityVal);
+          }
+        }
+
+        return std::make_shared<RowVector>(
+            PyVeloxContext::getSingletonInstance().pool(),
+            rowType,
+            nullabilityBuffer,
+            vectorSize,
+            values);
+      },
+      py::arg("children"),
+      py::arg("values"),
+      py::arg("nulls") = std::nullopt);
+
+  py::class_<RowVector, BaseVector, RowVectorPtr>(
+      m, "RowVector", py::module_local(asModuleLocalDefinitions))
+      .def("__len__", &RowVector::childrenSize)
+      .def("__str__", [](RowVectorPtr& v) { return rowVectorToString(v); })
+      .def("may_have_nulls", &RowVector::mayHaveNulls)
+      .def("__getitem__", [](RowVectorPtr& v, vector_size_t idx) {
+        return getVectorFromRowVectorPtr(v, idx);
+      });
```

Comment:
Also lets add `childAt` method to return vector at idx'i' . 
Should we also consider rowvector like dictionary ; so if I have row('a':<somevector>, 'b':<someothervector>), then row['a'] gives me <somevector>.